homeassistant/components/radarr/coordinator.py

```diff
@@ -2,27 +2,50 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from datetime import timedelta
+import asyncio
+from dataclasses import dataclass
+from datetime import date, datetime, timedelta
 from typing import Generic, TypeVar, cast
 
-from aiopyarr import Health, RadarrMovie, RootFolder, SystemStatus, exceptions
+from aiopyarr import (
+    Health,
+    RadarrCalendarItem,
+    RadarrMovie,
+    RootFolder,
+    SystemStatus,
+    exceptions,
+)
 from aiopyarr.models.host_configuration import PyArrHostConfiguration
 from aiopyarr.radarr_client import RadarrClient
 
+from homeassistant.components.calendar import CalendarEvent
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryAuthFailed
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import DEFAULT_MAX_RECORDS, DOMAIN, LOGGER
 
-T = TypeVar("T", bound=SystemStatus | list[RootFolder] | list[Health] | int)
+T = TypeVar("T", bound=SystemStatus | list[RootFolder] | list[Health] | int | None)
+
+
+@dataclass
+class RadarrEventMixIn:
```

Comment:
By using `kw_only=True` in the `dataclass` decorator we can avoid the mixin for required attributes.